File: SysGestionTurismo/src/main/java/pe/edu/upeu/sysgestionturismo/modelo/Hospedaje.java
package pe.edu.upeu.sysgestionturismo.modelo;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
public class Hospedaje {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idHospedaje;

    private String nombre;
    private String descripcion;
    private String direccion;
    private String whatsappContacto;
    private Double precioPorNoche;
    private String imagenPath;

    @ManyToOne
    @JoinColumn(name = "idDestino")
    private Destino destino;
}

File: SysTurismoJPC/app/src/main/java/pe/edu/upeu/systurismojpc/modelo/Hospedaje.kt
package pe.edu.upeu.systurismojpc.modelo

data class HospedajeDto(
    var idHospedaje: Long,
    var nombre: String,
    var descripcion: String,
    var direccion: String,
    var whatsappContacto: String,
    var precioPorNoche: Double,
    var imagenPath: String,
    var idDestino: Long
)

data class HospedajeSimpleDto(
    var idHospedaje: Long
)

data class HospedajeResp(
    val idHospedaje: Long,
    val nombre: String,
    val descripcion: String,
    val direccion: String,
    val whatsappContacto: String,
    val precioPorNoche: Double,
    val imagenPath: String,
    val idDestino: Long,
    val destinoNombre: String? = null 
)

fun HospedajeResp.toDto(): HospedajeDto {
    return HospedajeDto(
        idHospedaje = this.idHospedaje,
        nombre = this.nombre,
        descripcion = this.descripcion,
        direccion = this.direccion,
        whatsappContacto = this.whatsappContacto,
        precioPorNoche = this.precioPorNoche,
        imagenPath = this.imagenPath,
        idDestino = this.idDestino
    )
}